app/src/main/java/rectangledbmi/com/pittsburghrealtimetracker/BusMapFragment.java

```diff
@@ -0,0 +1,1002 @@
+            return;
+        }
+        Location lastLocation = LocationServices.FusedLocationApi.getLastLocation(googleApiClient);
+        if (lastLocation != null) { // use the last location if found
+            LatLng latLng = new LatLng(lastLocation.getLatitude(), lastLocation.getLongitude());
+            mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, zoomStopVisibility));
+        } else { // request one location update
+            LocationRequest gLocationRequest = LocationRequest.create()
+                    .setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY)
+                    .setInterval(1000)
+                    .setExpirationTime(10000) // set expiration 10 seconds
+                    .setNumUpdates(1); // only do one update. needs above call for expiration
+            LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, gLocationRequest, this);
+        }
+        enableGoogleMapLocation();
+    }
+
+    /**
+     * Enables the google map location UI settings if the permission is allowed.
+     */
+    private void enableGoogleMapLocation() {
+        if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, CENTER_MAP_LOCATION_CODE);
+            return;
+        }
+        mMap.setMyLocationEnabled(true);
+    }
+
+    /**
+     * Centers the map on the user's latest location data.
+     * @param location the user's latest location data.
+     */
+    @Override
+    public void onLocationChanged(Location location) {
+        /*
+        in the future, we may want different behaviors when the FusedLocationApi
+        requests location updates. If this needs to happen, we should have
+        multiple internal classes that implement LocationListener with different logic in each
+         */
+        if (location == null || mMap == null || isInPittsburgh(location)) return;
+        LatLng latLng = new LatLng(location.getLatitude(), location.getLongitude());
+        mMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, zoomStopVisibility));
+    }
+    // endregion
+
+    // region Map State and Observables
+
+    /**
+     * Resets the composite subscription for vehicles
+     */
+    private void resetVehicleSubscriptions() {
+        Timber.d("Resetting vehicle subscriptions in bus fragment");
+        if(vehicleSubscriptions == null || vehicleSubscriptions.isUnsubscribed()) {
+            vehicleSubscriptions = new CompositeSubscription(
+                    vehicleUpdateObservable.subscribe(vehicleUpdateObserver()),
+                    vehicleErrorObservable.subscribe(vehicleErrorObserver())
+            );
+            selectionSubject.onNext(RouteSelection.create(busListInteraction.getSelectedRoutes()));
```

Comment:
This is definitely being called, but it's not emitting anything when I start the app...... Is this a bug with subjects or what... could be related to the subject being crossed through fragments by a getter? I highly doubt it but still it's annoying. Log statements seem to tell me it's because of hte observable though.